File: 프로그래머스/lv2/12953. N개의 최소공배수/N개의 최소공배수.java
public class Solution {
	public int solution(int[] arr) {
		int answer = cld(arr[0],arr[1]);
		for(int i = 2; i < arr.length;i++) {
			answer = cld(answer, arr[i]);
		}
		return answer;

	}

	public int gcd(int a, int b) {
		while (b != 0) {
			int n = a % b;
			a = b;
			b = n;
		}

		return a;

	}
	
	public int cld(int a, int b) {
		return (a*b)/gcd(a,b);
	}
}

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.swift
func solution(_ arr:[Int]) -> Int {
    var ans = 1
    for i in 0 ..< arr.count {
        ans = lcm(ans, arr[i])
    }
    
    return ans
}

func gcd(_ a: Int, _ b: Int) -> Int {
    if b == 0 {
        return a
    } else {
        return gcd(b, a % b)
    }
}

func lcm(_ a: Int, _ b: Int) -> Int {
    return a * b / gcd(a, b)
}